packages/docusaurus-theme-classic/src/themeConfigSchema.js

```diff
@@ -0,0 +1,168 @@
+  position: NavbarItemPosition,
+  label: Joi.string(),
+  to: Joi.string(),
+  docsPluginId: Joi.string(),
+});
+
+const DocsVersionDropdownNavbarItemSchema = Joi.object({
+  type: Joi.string().equal('docsVersionDropdown').required(),
+  position: NavbarItemPosition,
+  docsPluginId: Joi.string(),
+});
+
+// Can this be made easier? :/
+const isOfType = (type) => {
+  let typeSchema = Joi.string().required();
+  // because equal(undefined) is not supported :/
+  if (type) {
+    typeSchema = typeSchema.equal(type);
+  }
+  return Joi.object({
+    type: typeSchema,
+  })
+    .unknown()
+    .required();
+};
+
+const NavbarItemSchema = Joi.object().when({
+  switch: [
+    {
+      is: isOfType('docsVersion'),
+      then: DocsVersionNavbarItemSchema,
+    },
+    {
+      is: isOfType('docsVersionDropdown'),
+      then: DocsVersionDropdownNavbarItemSchema,
+    },
+    {
+      is: isOfType(undefined),
+      then: Joi.forbidden().messages({
+        'any.unknown': 'Bad nav item type {.type}',
+      }),
+    },
+  ],
+  otherwise: DefaultNavbarItemSchema,
+});
+
+/*
+const NavbarItemSchema = Joi.object({
+  type: Joi.string().only(['docsVersion'])
+})
+  .when(Joi.object({ type: 'docsVersion' }).unknown(), {
+    then: Joi.object({ pepperoni: Joi.boolean() })
+  })
+  .when(Joi.object().unknown(), {
+    then: Joi.object({ croutons: Joi.boolean() })
+  })
+
+ */
+
+/*
+const NavbarItemSchema = Joi.object().when('type', {
+  is: Joi.valid('docsVersion'),
+  then: DocsVersionNavbarItemSchema,
+  otherwise: DefaultNavbarItemSchema,
+});
+ */
+
+const ColorModeSchema = Joi.object({
+  defaultMode: Joi.string().equal('dark', 'light').default('light'),
+  disableSwitch: Joi.bool().default(false),
+  respectPrefersColorScheme: Joi.bool().default(false),
+}).default({
+  defaultMode: 'light',
+  disableSwitch: false,
+  respectPrefersColorScheme: false,
+});
+
+const ThemeConfigSchema = Joi.object({
+  // TODO temporary (@alpha-58)
+  disableDarkMode: Joi.any().forbidden(false).messages({
+    'any.unknown':
+      'disableDarkMode theme config is deprecated. Please use the new colorMode attribute. You likely want: config.themeConfig.colorMode.disableSwitch = true',
+  }),
+  // TODO temporary (@alpha-58)
+  defaultDarkMode: Joi.any().forbidden(false).messages({
+    'any.unknown':
+      'defaultDarkMode theme config is deprecated. Please use the new colorMode attribute. You likely want: config.themeConfig.colorMode.defaultMode = "dark"',
+  }),
+  colorMode: ColorModeSchema,
+  image: Joi.string(),
+  announcementBar: Joi.object({
+    id: Joi.string(),
+    content: Joi.string(),
+    backgroundColor: Joi.string().default('#fff'),
+    textColor: Joi.string().default('#000'),
+  }).optional(),
+  navbar: Joi.object({
+    hideOnScroll: Joi.bool().default(false),
+    // TODO temporary (@alpha-58)
+    links: Joi.any().forbidden().messages({
+      'any.unknown':
+        'themeConfig.navbar.links has been renamed as themeConfig.navbar.items',
```

Comment:
This is awesome

Reply:
Thanks 😊

The renaming is not required but I think items makes more sense than links, now that navbar items have custom types. 

I imagine we'd have more navbar item types in the future, including custom ones provided by the user, and maybe the ability for plugins to provide some